Keycloak SAML Adapter subsystem does not work in WildFly 37

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

adapter/jee-saml

### Describe the bug

Attempting to use the latest Keycloak SAML adapter with the [newly released WildFly 37](https://www.wildfly.org/news/2025/08/04/WildFly-37-is-released/) results in a server start failure due to a missing constructor exception.

`Caused by: java.lang.NoSuchMethodError: 'void org.jboss.as.server.deployment.module.ModuleDependency.<init>(org.jboss.modules.ModuleLoader, org.jboss.modules.ModuleIdentifier, boolean, boolean, boolean, boolean)'
  at org.keycloak.keycloak-saml-wildfly-subsystem@26.3.2//org.keycloak.subsystem.adapter.saml.extension.KeycloakDependencyProcessorWildFly.addCoreModules(KeycloakDependencyProcessorWildFly.java:41)
  at org.keycloak.keycloak-saml-wildfly-subsystem@26.3.2//org.keycloak.subsystem.adapter.saml.extension.KeycloakDependencyProcessor.deploy(KeycloakDependencyProcessor.java:71)
  at org.jboss.as.server@29.0.0.Final//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:165)`

Upon investigation, the Keycloak adapter currently relies on deprecated & marked for removal features from WildFly core, some of which were removed in the latest WildFly core release, 29.0.0.Final. Namely, `org.jboss.modules.ModuleIdentifier` and the `org.jboss.as.server.deployment.module.ModuleDependency` constructor in use in `KeycloakDependencyProcessor `& `KeycloakDependencyProcessorWildFly` with in the `keycloak-saml-wildfly-subsystem` jar.

### Version

SAML Adapter 26.3.2

### Regression

- [ ] The issue is a regression

### Expected behavior

A SAML adapter configured WildFly instance to start successfully and allow for successful SAML authentication for the deployed & configured applications.

### Actual behavior

WildFly fails to start the SAML adapter and therefore, fails to start the whole server.

### How to Reproduce?

- Configure WildFly 37 with the latest SAML adapter, e.g. through installing the feature pack using galleon
- Configure & enable the keycloak-saml subsystem and elytron as required
- Attempt to start the server

### Anything else?

_No response_

--- a/adapters/saml/wildfly/wildfly-subsystem/src/main/java/org/keycloak/subsystem/adapter/saml/extension/KeycloakDependencyProcessor.java
+++ b/adapters/saml/wildfly/wildfly-subsystem/src/main/java/org/keycloak/subsystem/adapter/saml/extension/KeycloakDependencyProcessor.java
@@ -28,7 +28,6 @@ import org.jboss.as.web.common.WarMetaData;
 import org.jboss.metadata.web.jboss.JBossWebMetaData;
 import org.jboss.metadata.web.spec.LoginConfigMetaData;
 import org.jboss.modules.Module;
-import org.jboss.modules.ModuleIdentifier;
 import org.jboss.modules.ModuleLoader;
 
 /**
@@ -36,10 +35,10 @@ import org.jboss.modules.ModuleLoader;
  */
 public abstract class KeycloakDependencyProcessor implements DeploymentUnitProcessor {
 
-    private static final ModuleIdentifier KEYCLOAK_JBOSS_CORE_ADAPTER = KeycloakSubsystemDefinition.KEYCLOAK_JBOSS_CORE_ADAPTER;
-    private static final ModuleIdentifier KEYCLOAK_CORE_ADAPTER = ModuleIdentifier.create("org.keycloak.keycloak-saml-adapter-core");
-    private static final ModuleIdentifier KEYCLOAK_API_ADAPTER = ModuleIdentifier.create("org.keycloak.keycloak-saml-adapter-api-public");
-    private static final ModuleIdentifier KEYCLOAK_COMMON = ModuleIdentifier.create("org.keycloak.keycloak-common");
+    static final String KEYCLOAK_JBOSS_CORE_ADAPTER = "org.keycloak.keycloak-jboss-adapter-core";
+    static final String KEYCLOAK_CORE_ADAPTER = "org.keycloak.keycloak-saml-adapter-core";
+    static final String KEYCLOAK_API_ADAPTER = "org.keycloak.keycloak-saml-adapter-api-public";
+    static final String KEYCLOAK_COMMON = "org.keycloak.keycloak-common";
 
     @Override
     public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitProcessingException {

--- a/adapters/saml/wildfly/wildfly-subsystem/src/main/java/org/keycloak/subsystem/adapter/saml/extension/KeycloakDependencyProcessorWildFly.java
+++ b/adapters/saml/wildfly/wildfly-subsystem/src/main/java/org/keycloak/subsystem/adapter/saml/extension/KeycloakDependencyProcessorWildFly.java
@@ -23,7 +23,6 @@ import org.jboss.as.server.deployment.DeploymentPhaseContext;
 import org.jboss.as.server.deployment.module.ModuleDependency;
 import org.jboss.as.server.deployment.module.ModuleSpecification;
 import org.jboss.modules.ModuleClassLoader;
-import org.jboss.modules.ModuleIdentifier;
 import org.jboss.modules.ModuleLoader;
 
 /**
@@ -33,8 +32,7 @@ import org.jboss.modules.ModuleLoader;
  */
 public class KeycloakDependencyProcessorWildFly extends KeycloakDependencyProcessor {
 
-    private static final ModuleIdentifier KEYCLOAK_CORE_ADAPTER = ModuleIdentifier.create("org.keycloak.keycloak-saml-adapter-core");
-    private static final ModuleIdentifier KEYCLOAK_ELYTRON_ADAPTER = ModuleIdentifier.create("org.keycloak.keycloak-saml-wildfly-elytron-adapter");
+    private static final String KEYCLOAK_ELYTRON_ADAPTER = "org.keycloak.keycloak-saml-wildfly-elytron-adapter";
 
     @Override
     protected void addCoreModules(ModuleSpecification moduleSpecification, ModuleLoader moduleLoader) {

--- a/adapters/saml/wildfly/wildfly-subsystem/src/main/java/org/keycloak/subsystem/adapter/saml/extension/KeycloakSubsystemDefinition.java
+++ b/adapters/saml/wildfly/wildfly-subsystem/src/main/java/org/keycloak/subsystem/adapter/saml/extension/KeycloakSubsystemDefinition.java
@@ -21,7 +21,8 @@ import org.jboss.as.controller.SimpleResourceDefinition;
 import org.jboss.as.controller.operations.common.GenericSubsystemDescribeHandler;
 import org.jboss.as.controller.registry.ManagementResourceRegistration;
 import org.jboss.as.controller.registry.RuntimePackageDependency;
-import org.jboss.modules.ModuleIdentifier;
+
+import static org.keycloak.subsystem.adapter.saml.extension.KeycloakDependencyProcessor.KEYCLOAK_JBOSS_CORE_ADAPTER;
 
 /**
  * Definition of subsystem=keycloak-saml.
@@ -31,7 +32,6 @@ import org.jboss.modules.ModuleIdentifier;
 public class KeycloakSubsystemDefinition extends SimpleResourceDefinition {
 
     static final KeycloakSubsystemDefinition INSTANCE = new KeycloakSubsystemDefinition();
-    static final ModuleIdentifier KEYCLOAK_JBOSS_CORE_ADAPTER = ModuleIdentifier.create("org.keycloak.keycloak-jboss-adapter-core");
 
     private KeycloakSubsystemDefinition() {
         super(KeycloakSamlExtension.SUBSYSTEM_PATH,
@@ -51,6 +51,6 @@ public class KeycloakSubsystemDefinition extends SimpleResourceDefinition {
     public void registerAdditionalRuntimePackages(ManagementResourceRegistration resourceRegistration) {
         // This module is required by deployment but not referenced by JBoss modules
         resourceRegistration.registerAdditionalRuntimePackages(
-                RuntimePackageDependency.required(KEYCLOAK_JBOSS_CORE_ADAPTER.getName()));
+                RuntimePackageDependency.required(KEYCLOAK_JBOSS_CORE_ADAPTER));
     }
 }

